code/core/src/docs-tools/argTypes/enhanceArgTypes.ts

```diff
@@ -12,10 +12,10 @@ export const enhanceArgTypes = <TRenderer extends Renderer>(
   } = context;
   const { extractArgTypes } = docs;
 
-  const extractedArgTypes = extractArgTypes && component ? extractArgTypes(component) : {};
-  const withExtractedTypes = extractedArgTypes
-    ? (combineParameters(extractedArgTypes, userArgTypes) as typeof userArgTypes)
-    : userArgTypes;
+  if (!extractArgTypes || !component) {
+    return userArgTypes;
+  }
 
-  return withExtractedTypes;
+  const extractedArgTypes = extractArgTypes(component);
+  return extractedArgTypes ? combineParameters(extractedArgTypes, userArgTypes) : userArgTypes;
```

Comment:
the outcome here is the same, I just refactored it slightly to make it more readable. Spent way too many brain cycles on figuring out what happened if `extractArgTypes` or `component` was `undefined`.